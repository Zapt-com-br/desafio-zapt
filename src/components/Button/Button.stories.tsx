import React from 'react';

import { Story, Meta } from '@storybook/react/types-6-0';
import Button, { ButtonProps } from '.';

export default {
  title: 'Generic/Button',
  component: Button,
} as Meta;

export const ButtonDefault: Story<ButtonProps> = args => <Button {...args} />;

ButtonDefault.args = {
  label: 'Criar um grupo',
};
