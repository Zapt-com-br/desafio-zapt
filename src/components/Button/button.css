.btn-main {
  min-width: 161px;
  height: 51px;
  cursor: pointer;
  color: #fff;
  border: none;
  border-radius: 4px;
  background-color: var(--blue);
  font-size: 14px;
}