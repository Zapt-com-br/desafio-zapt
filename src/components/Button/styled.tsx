import styled from 'styled-components'

export const Button = styled.button`
  border: none;
  background-color: #334bc8;
  border-radius: 4px;
  color: #fff;

  padding: 1rem 2rem;

  transition: background-color 0.4s;

  &:hover {
    background: #ff5b58;
  }
`
