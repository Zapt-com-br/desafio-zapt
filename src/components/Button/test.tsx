import { render, screen } from 'utils/test-utils'

import Button from '.'

describe('<Button />', () => {
  it('should render the button with style and children', () => {
    render(<Button>Crie um grupo</Button>)

    expect(
      screen.getByRole('button', { name: /Crie um grupo/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('button', { name: /Crie um grupo/i })).toHaveStyle({
      'font-size': '1.4rem',
      'font-weight': '400',
      color: '#FFFFFF',
      'background-color': '#334BC8',
      padding: '1.6rem 3.2rem 1.8rem'
    })
  })
})
