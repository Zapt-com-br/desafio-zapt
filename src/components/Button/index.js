import React from 'react';
import './button.css';

function Button({ title }) {
  return (
    <button className="btn-main">{title}</button>
  );
}

export default Button;