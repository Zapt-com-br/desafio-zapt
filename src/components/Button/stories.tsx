import { Story } from '@storybook/react'
import {
  BoxArrowLeft,
  CheckCircle,
  ExclamationTriangle,
  GearFill,
  InfoCircle,
  PencilSquare,
  Save,
  Trash,
  X,
  XCircle
} from '@styled-icons/bootstrap'
import styled from 'styled-components'
import Button, { BtnProps } from '.'

const WrapperBtn = styled.div`
  display: flex;
  gap: 2rem;
  flex-wrap: wrap;
  justify-content: center;
`

export const Template: Story<BtnProps> = args => (
  <WrapperBtn>
    <Button {...args} />
    <Button {...args} icon={<Save />} />
    <Button {...args} text="" icon={<GearFill />} />
    <Button {...args} text="" icon={<Save />} />
    <Button {...args} text="" icon={<PencilSquare />} />
    <Button {...args} text="" icon={<Trash />} />
    <Button {...args} text="" icon={<X />} />
    <Button {...args} text="" icon={<InfoCircle />} />
    <Button {...args} text="" icon={<ExclamationTriangle />} />
    <Button {...args} text="" icon={<CheckCircle />} />
    <Button {...args} text="" icon={<XCircle />} />
    <Button {...args} text="" icon={<BoxArrowLeft />} />
  </WrapperBtn>
)

export const Default = Template.bind({})
