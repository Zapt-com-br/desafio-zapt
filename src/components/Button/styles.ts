import styled, { css } from 'styled-components';

export const Wrapper = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.primaryLight};
    padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
    border: 0;
    border-radius: ${theme.border.radius};
    cursor: pointer;
  `}
`;
