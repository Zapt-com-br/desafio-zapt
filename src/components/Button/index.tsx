import React, { ButtonHTMLAttributes } from 'react';
import * as Styled from './styles';

import { Typography } from 'components';

export type ButtonProps = {
  label: string;
  onClick?: () => void;
} & ButtonHTMLAttributes<HTMLButtonElement>;

const Button = ({ label, onClick }: ButtonProps) => (
  <Styled.Wrapper onClick={onClick}>
    <Typography color="white" size="small">
      {label}
    </Typography>
  </Styled.Wrapper>
);

export default Button;
