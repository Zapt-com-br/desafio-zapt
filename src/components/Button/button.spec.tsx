import React from 'react';
import { screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { renderWithTheme } from 'helpers/renderWithThame';

import Button from '.';

describe('<Button />', () => {
  it('should render button with label', () => {
    renderWithTheme(<Button label="button label" />);

    expect(screen.getByText(/button label/i)).toBeInTheDocument();
    expect(screen.getByText(/button label/i)).toMatchSnapshot();
  });

  it('should render button with on click if passed', async () => {
    const onclick = jest.fn();
    renderWithTheme(<Button label="button label" onClick={onclick} />);

    userEvent.click(screen.getByText(/button label/i));
    await waitFor(() => {
      expect(onclick).toHaveBeenCalledTimes(1);
    });
  });
});
