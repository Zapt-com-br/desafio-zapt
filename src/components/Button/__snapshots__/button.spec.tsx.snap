// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render button with label 1`] = `
.c0 {
  font-weight: 400;
  color: #ffffff;
  font-size: 1.4rem;
}

<span
  class="c0"
  color="white"
>
  button label
</span>
`;
